test/02_gsn_acl_factory.js

```diff
@@ -0,0 +1,117 @@
+chai.use(require('chai-as-promised'));
+const assert = chai.assert;
+
+const { fundRecipient } = require('@openzeppelin/gsn-helpers');
+const { fromConnection, useEphemeralKey } = require('@openzeppelin/network');
+const { utils } = require('@openzeppelin/gsn-provider');
+const { isRelayHubDeployedForRecipient, getRecipientFunds } = utils;
+
+const ACLFactory = artifacts.require("AclFactory");
+const ACL = artifacts.require("Acl");
+
+contract('AclFactory', (accounts) => {
+  const ROOT_ACCOUNT = process.env.NETWORK === 'ganache' ? accounts[0] : process.env.ACCOUNT;
+  const RELAY_HUB = process.env.RELAYHUB;
+  // Factory will be able to create 5 new individual ACLs
+  const GAS_PRICE = '500000000000';
+  const FACTORY_ACL_FAUCET_FUNDING_ETH = '50';
+  const FACTORY_FUNDING_ETH = '8';
+
+  let gsnCtx;
+  let emptyAcc;
+  let readerAcc;
+  let factory;
+  let newACLGSN;
+
+  it('should deploy a factory', async () => {
+    factory = await ACLFactory.new();
+
+    emptyAcc = await web3.eth.accounts.create("secret");
+    readerAcc = await web3.eth.accounts.create("secret");
+  });
+
+  it('should initialize the GSN for ACLFactory by configuring its RelayHub and funding it', async () => {
+    let txHub = await factory.initialize(RELAY_HUB);
+    assert.equal(txHub.receipt.status, true);
+
+    let hubAddr = await factory.getHubAddr();
+    assert.equal(hubAddr, RELAY_HUB);
+
+    const factoryAddr = await factory.address;
+
+    // Required so we can fund individual ACLs from ACLFactory automatically
+    const faucetTx = await web3.eth.sendTransaction({
+      from: ROOT_ACCOUNT,
+      to: factory.address,
+      value: web3.utils.toWei(FACTORY_ACL_FAUCET_FUNDING_ETH, "ether")
+    });
+    assert.equal(faucetTx.status, true);
+
+    // Required to register the Recipient in RelayHub
+    const balance = await fundRecipient(web3, {
+      recipient: factoryAddr,
+      relayHubAddress: RELAY_HUB,
+      amount: web3.utils.toWei(FACTORY_FUNDING_ETH, "ether"),
+      from: ROOT_ACCOUNT
+    });
+
+    assert.equal(balance, web3.utils.toWei(FACTORY_FUNDING_ETH, "ether"));
+  });
+
+  it('should deploy an ACL from a user without any funds for FREE using ACLFactory', async () => {
+    const isFactoryDeployed = await isRelayHubDeployedForRecipient(web3, factory.address);
+    assert.equal(isFactoryDeployed, true);
+
+    const factoryBalance = await getRecipientFunds(web3, factory.address);
+    assert.equal(factoryBalance, web3.utils.toWei(FACTORY_FUNDING_ETH, "ether"));
+
+    gsnCtx = await fromConnection(
+      web3.eth.currentProvider.host, {
+        gsn: {
+          dev: false,
+          signKey: emptyAcc.privateKey
+        }
+    });
+
+    web3 = gsnCtx.lib;
+    const factoryGSN = await new gsnCtx.lib.eth.Contract(factory.abi, factory.address);
+
+    const createNewACLRes = await factoryGSN.methods.newACL(emptyAcc.address).send({
+      from: emptyAcc.address,
+      gasPrice: GAS_PRICE,
+      gasLimit: "7000000",
```

Comment:
[Minor] Aiming a better scalability of this test in case we modify the complexity of the `newAcl()`, please calculate the estimatedGas using web3 using its api https://web3js.readthedocs.io/en/v1.2.0/web3-eth-contract.html#methods-mymethod-estimategas

Reply:
Tried right now, doesn't work properly. I am keeping the hardcoded value at least we know the upper limit. Later we can revisit by optimizing the compiler and so on.